docs/core/testing/unit-testing-mstest-migration-from-v1-to-v3.md

```diff
@@ -0,0 +1,290 @@
+
+---
+
+## Deprecated Features
+
+- Dropped support for:
+  - **.NET Framework 4.5** (use .NET 4.6.2 or higher)
+  - **.NET Standard 1.0** (use .NET Standard 2.0)
+  - **UWP versions before 16299**
+  - **WinUI versions before 18362**
+  - **.NET 5** (use .NET Core 3.1 or .NET 6)
+
+---
+
+## Breaking Changes and Removed APIs
+
+### Assertion Overloads
+
+MSTest v3 removes certain `Assert` overloads that accept object types to promote type-safe assertions. Update tests using these overloads to specify types explicitly.
+
+**Example**:
+
+```csharp
+// Old (v0/v1)
+Assert.AreEqual(expected, actual); // both are objects
+
+// New (v3)
+Assert.AreEqual<int>(expectedInt, actualInt); // specify the type
+```
+
+### DataRowAttribute Updates
+
+The `DataRowAttribute` constructors have been simplified in MSTest v3. Update parameterized tests using `DataRow` to align with the revised constructors.
+
+**Example**:
+
+```csharp
+[TestMethod]
+[DataRow(1, "test")]
+public void MyTestMethod(int number, string text) { ... }
+```
+
+### Timeout Settings
+
+In MSTest v3, the handling of `Timeout` settings has been standardized to ensure consistent behavior across different .NET environments. This change may impact tests that rely on specific timeout values, especially if those tests are asynchronous or run under different frameworks.
+
+- In MSTest v1 or v2, certain timeout settings might have been interpreted differently depending on the framework (e.g., .NET Framework vs. .NET Core).
+- MSTest v3 enforces consistent timeout behavior, which might mean that tests configured with timeouts in previous versions may fail or behave differently if the timeout values are too short under the new standard.
+
+**What This Means**:
+
+1. **Tests with Timeouts Might Need Adjustment**: If your tests have a `Timeout` attribute with specific durations, verify that those values still allow the test to complete under MSTest v3. Tests that previously passed with a certain timeout might need a higher or lower timeout value to work correctly under the new rules.
+
+2. **Unified Timeout Handling**: MSTest v3 unified timeout handling makes timeouts more predictable, but requires checking and potentially updating `Timeout` values in older tests.
+
+**Example**:
+
+```csharp
+// Old (v1/v2) - Timeout was sometimes interpreted inconsistently
+[TestMethod]
+[Timeout(2000)] // Timeout in milliseconds
+public void TestMethod() { ... }
+
+// New (v3) - Unified handling of timeout
+[TestMethod]
+[Timeout(2000)] // Verify this value still works under MSTest v3
+public async Task TestMethod() { ... } 
+
+---
+
+## Configuration Changes
+
+### Settings and Configuration Files
+
+MSTest v3 supports both XML and JSON formats for configuration files.
+
+1. **Verify Configurations**: Ensure that existing `.runsettings` files align with MSTest v3 syntax and structure.
+2. **Convert to JSON (if applicable)**: For JSON-based configurations, convert XML settings to JSON format.
```

Comment:
JSON is only supported by MSTest runner. I think we should focus on having users migrating to v3 first and have a special consideration section near the end to consider moving to the new platform.